unittest/client/src/ilm_control_unittests.cpp

```diff
@@ -0,0 +1,5071 @@
+            custom_wl_list_insert(&ilm_context.wl.list_surface, &mp_ctxSurface[i]->link);
+            //prepare the layers
+            mp_ctxLayer[i] = (struct layer_context*)malloc(sizeof(struct layer_context));
+            mp_ctxLayer[i]->id_layer = mp_ilmLayerIds[i];
+            mp_ctxLayer[i]->ctx = &ilm_context.wl;
+            mp_ctxLayer[i]->prop = mp_layerProps[i];
+            mp_ctxLayer[i]->notification = NULL;
+            custom_wl_list_insert(&ilm_context.wl.list_layer, &mp_ctxLayer[i]->link);
+            custom_wl_array_init(&mp_ctxLayer[i]->render_order);
+            // prepare the screens
+            mp_ctxScreen[i] = (struct screen_context*)malloc(sizeof(struct screen_context));
+            mp_ctxScreen[i]->id_screen = mp_ilmScreenIds[i];
+            mp_ctxScreen[i]->name = i;
+            mp_ctxScreen[i]->ctx = &ilm_context.wl;
+            mp_ctxScreen[i]->prop = mp_screenProps[i];
+            custom_wl_list_insert(&ilm_context.wl.list_screen, &mp_ctxScreen[i]->link);
+            custom_wl_array_init(&mp_ctxScreen[i]->render_order);
+        }
+    }
+
+    void deinit_ctx_list_content()
+    {
+        {
+            struct surface_context *l, *n;
+            wl_list_for_each_safe(l, n, &ilm_context.wl.list_surface, link)
+            {
+                custom_wl_list_remove(&l->link);
+            }
+        }
+        {
+            struct layer_context *l, *n;
+            wl_list_for_each_safe(l, n, &ilm_context.wl.list_layer, link)
+            {
+                custom_wl_list_remove(&l->link);
+            }
+        }
+        {
+            struct screen_context *l, *n;
+            wl_list_for_each_safe(l, n, &ilm_context.wl.list_screen, link)
+            {
+                custom_wl_list_remove(&l->link);
+            }
+        }
+        {
+            struct seat_context *l, *n;
+            wl_list_for_each_safe(l, n, &ilm_context.wl.list_seat, link)
+            {
+                custom_wl_list_remove(&l->link);
+            }
+        }
+        for(uint8_t i = 0; i < MAX_NUMBER; i++)
+        {
+            if(mp_ctxSurface[i] != nullptr)
+            {
+                free(mp_ctxSurface[i]);
+            }
+            if(mp_ctxLayer[i] != nullptr)
+            {
+                free(mp_ctxLayer[i]);
+            }
+            if(mp_ctxScreen[i] != nullptr)
+            {
+                free(mp_ctxScreen[i]);
+            }
+            if(mp_accepted_seat[i] != nullptr)
+            {
+                free(mp_accepted_seat[i]->seat_name);
+                free(mp_accepted_seat[i]);
+            }
+        }
+        for(uint8_t i = 0; i < 3; i++)
+        {
+            if(mp_ctxSeat[i] != nullptr)
+            {
+                free(mp_ctxSeat[i]->seat_name);
+                free(mp_ctxSeat[i]);
+            }
+        }
+        ilm_context.wl.controller = nullptr;
+        ilm_context.wl.notification = nullptr;
+        ilm_context.initialized = false;
```

Comment:
as discussed we should consider resetting the ilm_context in tear-down 
        memset(&ilm_context, 0, sizeof(ilm_context));